packages/legacy-client/src/client/ontology.ts

```diff
@@ -34,3 +43,20 @@ export class Ontology<O extends OntologyDefinition<any>> {
     throw new Error("not implemented");
   }
 }
+
+export function createObjectSetCreator<
+  O extends OntologyDefinition<any>,
+>(clientContext: ClientContext, definition: O): Objects<O> {
+  return new Proxy(
+    {},
+    {
+      get: (_target, p, _receiver) => {
+        if (typeof p === "string") {
+          return createBaseOsdkObjectSet(clientContext, p, definition);
+        }
+
+        return undefined;
+      },
+    },
+  ) as Objects<O>;
```

Comment:
As an alternate implementation you can skip the `Proxy` and just make an interface with a bunch of methods on it based on `definition`. Proxy is slower, but I'm also not sure that that super matters in the case where we're creating a whole object set vs hydrating a bunch of objects from the wire.